src/App.php

```diff
@@ -1158,4 +1169,31 @@ private function getRenderedPortals(): array
 
         return $portals;
     }
+
+    protected function emitResponse(): void
```

Comment:
```suggestion
    /**
     * @return never
     */
    protected function emitResponse(): void
```

If this method is expected to be terminating, use this phpdoc and put `exit;` at the end of the method

Reply:
IMHO we cannot exit like this, this method only output the response but the app must terminate as before, with stop execution or catching ExitApplicationError in callExit, called by App::run() or App::terminate(), if we exit there HOOK_BEFORE_EXIT is not fired and in other context will break the handler flow.

What you think about making it private and add a boolean flag to App like $emit_response?

Reply:
I see, never can not be used...